lightningd/peer_control.c

```diff
@@ -68,10 +68,15 @@ void peer_fail(struct peer *peer, const char *fmt, ...)
 		peer_reconnect(peer);
 }
 
-void peer_set_condition(struct peer *peer, enum peer_state state)
+void peer_set_condition(struct peer *peer, enum peer_state old_state,
+			enum peer_state state)
 {
 	log_info(peer->log, "state: %s -> %s",
 		 peer_state_name(peer->state), peer_state_name(state));
+	if (peer->state != old_state)
+		fatal("peer state %s should be %s",
```

Comment:
Yes, we should really review *all* calls to assert(), fatal() etc in the main daemon.  Probably come up with a "SHOULD_NOT_HAPPEN()" macro which we can make fatal in dev mode.  Will note.